zipkin-storage/cassandra-v1/src/test/java/zipkin2/storage/cassandra/v1/CassandraStorageExtension.java

```diff
@@ -109,54 +93,27 @@ InetSocketAddress contactPoint() {
       // Only run once in outermost scope.
       return;
     }
-
-    try {
-      if (session != null) session.close();
-      if (cluster != null) cluster.close();
-    } catch (Exception | Error e) {
-      LOGGER.warn("error closing session " + e.getMessage(), e);
-    } finally {
-      if (container != null) {
-        LOGGER.info("Stopping docker image " + image);
-        container.stop();
-      }
-    }
+    if (globalSession != null) globalSession.getCluster().close();
   }
 
   static final class CassandraContainer extends GenericContainer<CassandraContainer> {
     CassandraContainer(String image) {
       super(image);
     }
 
-    @Override
-    protected void waitUntilContainerStarted() {
-      Unreliables.retryUntilSuccess(
-        120,
-        TimeUnit.SECONDS,
-        () -> {
-          if (!isRunning()) {
-            throw new ContainerLaunchException("Container failed to start");
-          }
-
-          InetSocketAddress address =
-            new InetSocketAddress(getContainerIpAddress(), getMappedPort(9042));
-
-          try (Cluster cluster = getCluster(address);
-               Session session = cluster.newSession()) {
-            session.execute("SELECT now() FROM system.local");
-            logger().info("Obtained a connection to container ({})", cluster.getClusterName());
-            return null; // unused value
-          }
-        });
+    @Override protected void waitUntilContainerStarted() {
+      Unreliables.retryUntilSuccess(120, TimeUnit.SECONDS, () -> {
+        if (!isRunning()) {
+          throw new ContainerLaunchException("Container failed to start");
+        }
+
+        String contactPoint = getContainerIpAddress() + ":" + getMappedPort(9042);
+        try (Session session = Access.tryToInitializeSession(contactPoint)) {
+          session.execute("SELECT now() FROM system.local");
+          logger().info("Obtained a connection to container ({})", contactPoint);
+          return null; // unused value
+        }
+      });
     }
   }
-
-  static Cluster getCluster(InetSocketAddress contactPoint) {
```

Comment:
might not be obvious, but these settings repeated in the default session factory, except partially. moving it up allows better control over what we configure and also things like the same logging config to work in test sessions as the main sessions